src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -141,94 +190,119 @@ public class Thread implements Runnable {
         registerNatives();
     }
 
-    private volatile String name;
-    private int priority;
-
-    /* Whether or not the thread is a daemon thread. */
-    private boolean daemon = false;
-
-    /* Interrupt state of the thread - read/written directly by JVM */
-    private volatile boolean interrupted;
-
-    /* Fields reserved for exclusive use by the JVM */
-    private boolean stillborn = false;
+    /* Reserved for exclusive use by the JVM, maybe move to FieldHolder */
     private long eetop;
 
-    /* What will be run. */
-    private Runnable target;
+    // thread id
+    private final long tid;
+
+    // thread name
+    private volatile String name;
 
-    /* The group of this thread */
-    private ThreadGroup group;
+    // interrupt status (read/written by VM)
+    volatile boolean interrupted;
 
-    /* The context ClassLoader for this thread */
-    private ClassLoader contextClassLoader;
+    // context ClassLoader
+    private volatile ClassLoader contextClassLoader;
 
-    /* The inherited AccessControlContext of this thread */
+    // inherited AccessControlContext, this could be moved to FieldHolder
     @SuppressWarnings("removal")
     private AccessControlContext inheritedAccessControlContext;
 
-    /* For autonumbering anonymous threads. */
-    private static int threadInitNumber;
-    private static synchronized int nextThreadNum() {
-        return threadInitNumber++;
+    // Additional fields for platform threads
+    private static class FieldHolder {
```

Comment:
All but the task field are explicitly known by the VM (and stillborn is not accessed by Java code. Suggest grouping accordingly.